modules/schedule_module/jsx/scheduleIndex.js

```diff
@@ -0,0 +1,592 @@
+      if (cell==='Complete' ) {
+        css='label label-success';
+       }
+      if (cell==='No Data Found' ) {
+        css='label label-danger';
+       }
+      if (cell==='In Progress' || cell==='Not Started') {
+        css='label label-warning';
+       }
+      result = <td><span className={css}>{cell}</span></td>;
+      break;
+    case 'Appointment Type':
+      result = <td>{row['AppointmentTypeName']}</td>;
+      break;
+    }
+    return result;
+}
+  renderScheduleFormButton() {
+   if (this.state.editModal) {
+     return (<ButtonElement
+            name="edit"
+            label={
+              <div>
+                Edit Appointment
+              </div>
+            }
+            type="submit"
+            buttonClass="btn btn-sm btn-success"
+           />
+     );
+    } else {
+        return (<ButtonElement
+            name="create"
+            label={
+              <div>
+                Create Appointment
+              </div>
+            }
+            type="submit"
+            buttonClass="btn btn-sm btn-success"
+           />
+         );
+    }
+  }
+  renderScheduleForm() {
+  let year = new Date();
+  let minYear = year.getFullYear();
+    const title = this.state.editModal ? 'Edit Appointment' : 'Add Appointment';
+    return (
+      <Modal
+        title= {title}
+        onClose={this.closeModal}
+        show={this.state.showModal}
+      >
+        <FormElement
+          Module="schedule"
+          name="addSchedule"
+          id="addScheduleForm"
+          onSubmit={this.handleSubmit}
+          method="POST"
+        >
+          <TextboxElement
+            name="DCCID"
+            label="DCCID"
+            value={this.state.formData.DCCID}
+            required={true}
+            onUserInput={this.setFormData}
+            disabled={this.state.editModal}
+          />
+          <TextboxElement
+            name="PSCID"
+            label="PSCID"
+            value={this.state.formData.PSCID}
+            required={true}
+            onUserInput={this.setFormData}
+            disabled={this.state.editModal}
+          />
+          <SelectElement
+            name="Session"
+            options={this.state.formData.SessionFieldOptions}
+            label="Visit"
+            value={this.state.formData.Session}
+            required={true}
+            onUserInput={this.setFormData}
+            disabled={this.state.editModal}
+          />
+          <DateElement
+            name = "AppointmentDate"
+            label = "Appointment Date"
+            onUserInput = {this.setFormData}
+            value = {this.state.formData.AppointmentDate}
+            minYear = {minYear}
+            required = {true}
+          />
+          <TimeElement
+            name = "AppointmentTime"
+            label = "Appointment Time"
+            onUserInput = {this.setFormData}
+            value = {this.state.formData.AppointmentTime}
+            required = {false}
```

Comment:
should this be required or the test plan item 1 update?

Reply:
I will update the test plan , it has a default time (12:00).